Source/UI/PaymentForm/View/ExpiryDateView.swift

```diff
@@ -1,201 +0,0 @@
-import UIKit
```

Comment:
@alex-ioja-yang-cko Do you know which line in the older logic was causing the issue on Intel machines? would be interested to know if this is a swift API issue and how it behaves differently in different machines as the new approach still uses all the TextFieldViewDelegate methods which I assume is required.

Reply:
[This](https://github.com/checkout/frames-ios/pull/393/files#diff-5a7fe3d7bbb6ad57fd1dd9577704228acf52cfae0bc0ed52e0747328a99e606cL97-L105) (might not open correctly, so after you open link in a new tab, make sure to again expand the deleted file).

***

It was assuming a linear input occurring, where we always only modify the last character in the String (appending or removing). This was already prone to invalid behaviour if cutting/pasting or moving the cursor to somewhere else in the String. But the difference in behaviour seems to be from the values of `Range` and the way the code was triggered 2 times for every input on the runners we could reproduce. 

***
The flow for that code on a normal behaviour was:
- input change => modification reported with a Range (index of change, length 0)

The flow when the error was encountered (could only reproduce on Intel runners, hence the assumption it may be related 🤷‍♂️) was:
- input change
        => modification reported with a Range (index of change, length 0) - as before
        => modification reported with a Range (index of change, length 1) - unexpected ❗ . Here, our code was assuming any length present in the range was a deletion of previous content, and since it expected incremental updates, it would run into a "character deletion routine" where the `/` was removed. And then following it by simply adding the next character. This resulted in the index skipping the number where it should add separator.